spec/requests/api/v2/providers/sync_courses_with_search_and_compare_spec.rb

```diff
@@ -83,18 +83,12 @@
           end
 
           it 'should make the appropriate request' do
-            subject
+            perform_enqueued_jobs do
+              subject
+            end
```

Comment:
#678 adds some extra tests to check that an actual request is made out by the job, what do you think about adding these here?

https://github.com/DFE-Digital/manage-courses-backend/pull/678/files#diff-eed410faa3db2404572332f7ea75e7b2R40-R47

I know that this is potentially mixing in integration testing with request specs, but it'd be pretty convenient to put them here and doesn't feel completely out of place. Also, if we don't do them in this PR we're not likely to add them anywhere else.